snakemake_executor_plugin_slurm/partitions.py

```
@@ -14,6 +14,35 @@
+        )
+    except subprocess.CalledProcessError as e:
+        raise WorkflowError(
+            f"Failed to run sinfo for retrieval of cluster partitions: {e.stderr}"
+        )
+    for partition in out.split():
+        # A default partition is marked with an asterisk, but this is not part of
+        # the name.
+        if "*" in partition:
+            # the decode-call is necessary, because the output of sinfo is bytes
+            return partition.replace("*", "")
+    self.logger.warning(
```

Ruff (0.15.1)

[error] Undefined name `self`

(F821)